### PR TITLE
[#1009] Remove managed dependencies from build descriptors

### PR DESCRIPTION
**Pull request by kthoms**
- removed managed versions of
  - org.eclipse.emf.mwe2.launch
  - xtext-antlr-generator
  - org.junit:junit
  - org.junit.jupiter.api
  - xtext-dev-bom
- updated test expectation
- GradleBuildFile: Use single quoted dependencies for managed
dependencies (where version is null)

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by kthoms** [comment]
Depends on https://github.com/eclipse/xtext-lib/pull/156

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.full/org.xtext.example.full.parent/build.gradle [hunk 1]
```diff
@@ -15,7 +15,7 @@ subprojects {
 	
 	apply plugin: 'java'
 	dependencies {
-		compile platform("org.eclipse.xtext:xtext-dev-bom:unspecified")
```
Comment (cdietrich) [inline org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.full/org.xtext.example.full.parent/build.gradle]: did you test that?

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/GradleBuildFile.xtend [hunk 2]
```diff
@@ -27,7 +27,11 @@ class GradleBuildFile extends TextFile {
 						compile project(':�p.name�')
 					�ENDFOR�
 					�FOR dep : mavenDependencies�
-						�dep.scope.gradleNotation� "�dep.groupId�:�dep.artifactId�:�dep.version�"
+						�IF dep.version !== null�
+							�dep.scope.gradleNotation� "�dep.groupId�:�dep.artifactId�:�dep.version�"
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/GradleBuildFile.xtend]: wonder if we just add the if for the 3rd part.
would find it better readable

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeProjectDescriptor.xtend [hunk 3]
```diff
@@ -280,10 +279,10 @@ class RuntimeProjectDescriptor extends TestedProjectDescriptor {
 				}
 
 				dependencies {
-					mwe2 "org.eclipse.emf:org.eclipse.emf.mwe2.launch:�config.xtextVersion.mweVersion�"
```
Comment (cdietrich) [inline org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeProjectDescriptor.xtend]: should the values in the config then be marked as deprecated or are they used elsewhere?

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.full/org.xtext.example.full.parent/build.gradle (hunk 1 above)
Comment (kthoms) [reply org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.full/org.xtext.example.full.parent/build.gradle]: for newly created projects, yes.

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/GradleBuildFile.xtend (hunk 2 above)
Comment (kthoms) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/GradleBuildFile.xtend]: Did you recognize that there is a difference in single vs double quote also?

File: org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.full/org.xtext.example.full.parent/build.gradle (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.tests/testdata/wizard-expectations/org.xtext.example.full/org.xtext.example.full.parent/build.gradle]: :+1:

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/GradleBuildFile.xtend (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/GradleBuildFile.xtend]: yes

Comment (kthoms) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/GradleBuildFile.xtend]:
```
					«FOR dep : mavenDependencies»
						«dep.scope.gradleNotation» «IF dep.version !== null»"«dep.groupId»:«dep.artifactId»:«dep.version»"«ELSE»'«dep.groupId»:«dep.artifactId»'«ENDIF»
					«ENDFOR»
```
is actually less readable, or what did you mean?

Comment (cdietrich) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/GradleBuildFile.xtend]: then lets go your way

File: org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeProjectDescriptor.xtend (hunk 3 above)
Comment (kthoms) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeProjectDescriptor.xtend]: good point. I'll check that. Likely they can be deprecated.

Comment (kthoms) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeProjectDescriptor.xtend]: It does not work to remove dependency versions from exec-maven-plugin :( So there versions stay mandatory.

Comment (kthoms) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeProjectDescriptor.xtend]: But we should pin xtext-antlr-generator to 2.1.1 and no open range. There is no other version and won't.

Comment (cdietrich) [reply org.eclipse.xtext.xtext.wizard/src/org/eclipse/xtext/xtext/wizard/RuntimeProjectDescriptor.xtend]: :+1:

**Comment by kthoms** [comment]
I think we are done here for a first step. If there is potential to use managed dependencies for Tycho builds is another thing.

**Comment by kthoms** [comment]
Which test projects did you mean? generateTestLanguages does not produce a change, the wizard expectations are updated.

**Comment by cdietrich** [comment]
reference projects

**Comment by kthoms** [comment]
ah ok, will do!

**Pull request merged**